node/core/backing/src/lib.rs

```diff
@@ -0,0 +1,1454 @@
+					report: v.clone(),
+					signing_context: self.table_context.signing_context.clone(),
+					key: self.table_context.validators[*k as usize].clone(),
+				};
+
+				if let Ok(report) = MisbehaviorReport::try_from(f) {
+					let message = ProvisionerMessage::ProvisionableData(
+						ProvisionableData::MisbehaviorReport(self.parent, report)
+					);
+
+					reports.push(message);
+				}
+			}
+		}
+
+		for report in reports.drain(..) {
+			self.send_to_provisioner(report).await?
+		}
+
+		Ok(())
+	}
+
+	async fn import_statement(&mut self, statement: SignedFullStatement) -> Result<(), Error> {
+		let idx = statement.validator_index() as usize;
+
+		if self.table_context.validators.len() > idx {
+			match statement.check_signature(
+				&self.table_context.signing_context,
+				&self.table_context.validators[idx],
+			) {
+				Ok(()) => {
+					let stmt = primitive_statement_to_table(&statement);
+
+					self.table.import_statement(&self.table_context, stmt);
+
+					self.issue_new_misbehaviors().await?;
+				}
+				Err(()) => {
+					return Err(Error::ValidationFailed(ValidationFailed));
+				}
+			}
+		};
+
+		Ok(())
+	}
+
+	async fn process_msg(&mut self, msg: CandidateBackingMessage) -> Result<(), Error> {
+		match msg {
+			CandidateBackingMessage::Second(_, candidate, pov) => {
+				// If the message is a `CandidateBackingMessage::Second`, sign and dispatch a
+				// Seconded statement only if we have not seconded any other candidate and
+				// have not signed a Valid statement for the requested candidate.
+				match self.seconded {
+					// This job has not seconded a candidate yet.
+					None => {
+						let candidate_hash = candidate.hash();
+
+						if !self.issued_validity.contains(&candidate_hash) {
+							if let Ok(ValidationResult::Valid) = self.validate_and_second(
+								candidate,
+								pov,
+							).await {
+								self.seconded = Some(candidate_hash);
+							}
+						}
+					}
+					// This job has already seconded a candidate.
+					Some(_) => {}
+				}
+			}
+			CandidateBackingMessage::Statement(_, statement) => {
+				self.maybe_validate_and_import(statement).await?;
+			}
+			CandidateBackingMessage::GetBackedCandidates(_, tx) => {
+				let backed = self.get_backed();
+
+				tx.send(backed).map_err(|_| oneshot::Canceled)?;
+			}
+		}
+
+		Ok(())
+	}
+
+	async fn maybe_validate_and_import(
+		&mut self,
+		statement: SignedFullStatement,
+	) -> Result<(), Error> {
+		let idx = statement.validator_index() as usize;
+
+		if self.table_context.validators.len() > idx {
+			if let Err(()) = statement.check_signature(
+				&self.table_context.signing_context,
+				&self.table_context.validators[idx],
+			) {
+				return Err(Error::InvalidSignature);
+			}
+		} else {
+			return Err(Error::InvalidSignature);
+		}
+
+		if let Statement::Seconded(candidate) = statement.payload() {
+			if candidate.parachain_index == self.assignment {
+				let pov = self.request_pov_from_distribution(candidate.clone().into()).await?;
```

Comment:
Check out what I did in #1344 : We don't need to `clone` the entire `AbridgedCandidateReceipt` (which can be large, containing the commitments) just to get a `CandidateDescriptor`.